contracts/cw-token-swap/src/msg.rs

```diff
@@ -0,0 +1,60 @@
+use cosmwasm_std::Uint128;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::state::CheckedCounterparty;
+
+/// Information about the token being used on one side of the escrow.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum TokenInfo {
+    /// A native token.
+    Native { denom: String, amount: Uint128 },
```

Comment:
everyday i pray for TokenFactory

Reply:
lol me too I think I've written this code like 100 times.